File: C/LIS.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
void LIS(int *, int);
int ceilindex(int A[], int l, int r, int key);
void printLIS(int T[], int n);
int main()
	{
	
	int A[] = {2, 5, 3, 7, 11, 8, 10, 13, 6};
	int size = sizeof(A)/sizeof(A[0]);
	LIS( A,  size);
	return 0;
	}
void printLIS(int T[], int n)
		{
			int i;
			for(i=0;i<n;i++)
			{
				printf("\t[%d]",T[i]);
			
			}
		
		
		}
int ceilindex(int A[], int l, int r, int key)
			{
				int m;
				
				while(r-l>1)
					{
						m= l + (r-l)/2;
						if(A[m]>=key)
						{
							r=m;
						}else{
						
						l=m;
						}
					
					}
			return r;
			
			
			
			}
void LIS(int A[], int n)
{
	int T[n];
	int len,i, c;
	memset(T,0,n*sizeof(int));
	T[0] = A[0];
	len = 1;
	for(i=1;i<n;i++)
		{
			if(A[i]<T[0])
				T[0] = A[i];
			else if(A[i]>T[len-1])
				{T[len] = A[i];
				len++;}
			else
				{
				c= ceilindex(T,-1, len-1,A[i]);
				T[c]=A[i];
				printf("ll\t");
				}
		
		}
	
	

printf("len of LIS:%d\n\n",len);
printLIS(T, len);

}	

File: GFG/Array/LIS.java
import java.io.*;
import java.lang.*;
import java.util.*;

// #include <stdio.h>
// #include <stdlib.h>
// #include <string.h>
// #include<stdbool.h>
/*

# Author			: @RAJ009F
# Topic or Type 	: GFG/ARRAY
# Problem Statement	: Sub Array with given sum
# Description		: 
# Complexity		: 
=======================
#sample output
----------------------

=======================
*/

class LIS
{
	
	public static  void lis(int arr[], int len)
	{
		int[] L = new int[len];
		
		for(int i=0; i<len; i++)
			L[i] = 1;
		for(int i=1; i<len; i++)
		{
			for(int j=0; j<i; j++)
			{
				if(arr[i]>arr[j] && L[i]<L[j]+1)
					L[i] = L[j]+1;
				
			}
			
		}
		int max = L[0];
		for(int i=1; i<len; i++)
			if(max<L[i])
				max = L[i];
	System.out.println(max);
		
		
	}
	
	public static void main(String args[])
	{
		
		int arr[] = { 10, 22, 9, 33, 21, 50, 41, 60 };
		lis(arr, arr.length);
		
		
	}
}